spec/workers/step_by_step_draft_update_worker_spec.rb

```diff
@@ -44,16 +44,26 @@
     context 'when the guide is a new draft' do
       it 'generates a note that says "Draft created by New author"' do
         described_class.new.perform(step_by_step_page.id, @current_user.name)
-        expect(step_by_step_page.internal_change_notes.first.description).to eql 'Draft created by New author'
+        expect(step_by_step_page.internal_change_notes.first.description).to eql 'Draft saved by New author'
       end
     end
     context 'when the guide is a draft that has already been updated' do
       before do
         step_by_step_page.mark_draft_updated
       end
-      it 'should not generate a change note' do
+      it 'should generate a change note' do
         described_class.new.perform(step_by_step_page.id, @current_user.name)
-        expect(step_by_step_page.internal_change_notes.count).to eql 0
+        expect(step_by_step_page.internal_change_notes.count).to eql 1
+      end
+    end
+    context "when it is updated by the same user" do
+      before do
+        step_by_step_page.mark_draft_updated
+      end
+      it "should not generate another change note" do
+        described_class.new.perform(step_by_step_page.id, @current_user.name)
```

Comment:
I think you use the factory to set the initial user: `create(:step_by_step_page, assigned_to: @current_user.name)`